htdocs/class/module.textsanitizer.php

```
@@ -536,13 +536,26 @@ public function addSlashes($text)
+     *
+     * Arrays and objects are still rejected, which is the check actually worth keeping.
+     *
+     * @param  string|int|float|null $text string being converted
      * @param  int|null    $quote_style
      * @param  string|null $charset character set used in conversion
      * @param  bool   $double_encode
      * @return string
+     * @since  2.7.3 widened from string
      */
-    public function htmlSpecialChars(string $text, ?int $quote_style = null, ?string $charset = null, $double_encode = true)
+    public function htmlSpecialChars(string|int|float|null $text, ?int $quote_style = null, ?string $charset = null, $double_encode = true)
```

PHPMD (2.15.0)

[error] The method htmlSpecialChars has a boolean flag argument $double_encode, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)